tensorboard/plugins/core/core_plugin_test.py

```diff
@@ -37,30 +39,58 @@ class CorePluginTest(tf.test.TestCase):
   _only_use_meta_graph = False  # Server data contains only a GraphDef
 
   def setUp(self):
-    self.logdir = self.get_temp_dir()
-    self.addCleanup(shutil.rmtree, self.logdir)
+    self.temp_dir = self.get_temp_dir()
+    self.addCleanup(shutil.rmtree, self.temp_dir)
+
+    self.startLogdirBasedServer(self.temp_dir)
+    self.startDbBasedServer(self.temp_dir)
+
+  def startLogdirBasedServer(self, temp_dir):
+    self.logdir = temp_dir
     self._generate_test_data(run_name='run1')
     self.multiplexer = event_multiplexer.EventMultiplexer(
         size_guidance=application.DEFAULT_SIZE_GUIDANCE,
         purge_orphaned_data=True)
-    self._context = base_plugin.TBContext(
+    context = base_plugin.TBContext(
         assets_zip_provider=get_test_assets_zip_provider(),
         logdir=self.logdir,
         multiplexer=self.multiplexer)
-    self.plugin = core_plugin.CorePlugin(self._context)
+    self.logdir_based_plugin = core_plugin.CorePlugin(context)
     app = application.TensorBoardWSGIApp(
-        self.logdir, [self.plugin], self.multiplexer, 0, path_prefix='')
-    self.server = werkzeug_test.Client(app, wrappers.BaseResponse)
+        self.logdir,
+        [self.logdir_based_plugin],
+        self.multiplexer,
+        0,
+        path_prefix='')
+    self.logdir_based_server = werkzeug_test.Client(app, wrappers.BaseResponse)
+
+  def startDbBasedServer(self, temp_dir):
+    self.db_uri = 'sqlite:' + os.path.join(temp_dir, 'db.sqlite')
```

Comment:
Can we use an in-memory SQLite DB as in `:memory:` rather than writing to disk?  I guess `application.create_sqlite_connection_provider()` doesn't support it and that's what `application.get_database_info()` calls... maybe we can ask Justine why that restriction exists?

Reply:
I added that if statement probably because it doesn't make logical sense to say `--db=sqlite::memory:`. We will need in-memory SQLite in cases where `--logdir` is passed, but not `--db`.

---

**Commentary unrelated to this change**

I'll likely send a PR soon changing the way we think about the DB flag. After carefully studying the SQLite C API, I think the concept of a "SQLite URI" is a mistake, for many reasons. 

Firstly, the SQLite C API, if one actually chooses to use its "URI API" it then uses `file:` as the scheme. Yet here and in many other frameworks, `sqlite:` or `sqlite3:` have been used as its scheme. I believe the URI scheme for SQLite should be empty/null.

Secondly, a plain old path IS a an RFC 3986 compliant URI. Proof:

```pycon
>>> import urlparse
>>> urlparse.urlparse('/hello')
ParseResult(scheme='', netloc='', path='/hello', params='', query='', fragment='')
```

Thirdly, I believe it will be far wiser for us to use environment variables for SQLite tuning, so long as they're only used for tuning. See our [sqlite.cc](https://github.com/tensorflow/tensorflow/blob/febdd26ae594133d24f82544706b1e012a5cf1ea/tensorflow/core/lib/db/sqlite.cc#L135).

Reply:
It'd be nice if we could use in-memory sqlite for tests like these, though.

Re: the URI, I'm fine with walking back that change, but won't we still want something in the `--db` flag value to indicate what type of database it is?  E.g. if we see the `sqlite:` prefix we know it's a local sqlite file, versus later when we might be passing like a hostname+IP to a remote DB, or even something like a spanner path under `/span/` (for internal users).

Reply:
In the short term, how about we keep forbidding `:memory:` since it doesn't make much sense for writers to write to an in-memory database. I think we might keep the test URI to be prefixed with `sqlite:` for now then to serve as an example of usage for others.

And yeah, the `sqlite` URI prefix is strange - it's not really a protocol, so not sure why it deserves a special prefix.